app/assets/javascripts/utils/media_wiki_references_utils.js

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable no-restricted-syntax */
+import { ORESSupportedWiki } from './article_finder_language_mappings';
+import { queryUrl } from './article_finder_utils';
+import { chunk } from 'lodash-es';
+import { getReferencesCount } from './revision_utils';
+
+import promiseLimit from 'promise-limit';
+
+const limit = promiseLimit(5);
+
+// given a particular wiki, find all the references added for each revision in it
+const fetchReferencesAddedFromWiki = async (wiki_url, revisions) => {
+  // eslint-disable-next-line no-console
+  console.log(`Fetching references information from ${wiki_url}`);
+
+  const list = wiki_url.split('.');
+  let wiki;
+  if (list.length === 3) {
+    wiki = { language: list[0], project: list[1] };
+  } else {
+    wiki = { project: list[0] };
+  }
+  if (
+    !ORESSupportedWiki.projects.includes(wiki.project)
+  && !ORESSupportedWiki.languages.includes(wiki.language)
+  ) {
+    // wiki does not support ORES
+    return;
+  }
+  let models;
+  let suffix;
+  if (wiki.project === 'wikidata') {
+    suffix = 'wikidatawiki';
+    models = 'itemquality';
+  } else {
+    suffix = `${wiki.language}wiki`;
+    models = 'articlequality';
+  }
+  const API_URL = `https://ores.wikimedia.org/v3/scores/${suffix}`;
+  const revids = revisions.filter(revision => revision.ns === 0).map(revision => `${revision.parentid}|${revision.revid}`);
```

Comment:
This and the following `chunks` could use some additional comments to explain what's going on with the magic numbers and the `parentid` values.